packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -518,23 +519,20 @@ export class GraphQLApi extends Construct {
     };
   }
 
-  private createAPIKey(config: ApiKeyConfig) {
-    let expires: number | undefined;
-    if (config.expires) {
-      expires = new Date(config.expires).valueOf();
-      const days = (d: number) =>
-        Date.now() + Duration.days(d).toMilliseconds();
-      if (expires < days(1) || expires > days(365)) {
-        throw Error('API key expiration must be between 1 and 365 days.');
-      }
-      expires = Math.round(expires / 1000);
+  private createAPIKey(config: ApiKeyConfig): string {
+    if (config.expires &&
+      (config.expires?.date < Expires.after(Duration.days(1)).date
+        || config.expires?.date > Expires.after(Duration.days(365)).date)) {
+      throw Error('API key expiration must be between 1 and 365 days.');
     }
+    const getEpoch = (d: Expires) => { return Math.floor( d.date.getTime() / 86400000) * 86400; };
```

Comment:
Add `expires.asEpoch()` perhaps?

Reply:
Eli commented before and said that it would be best to move this function out of the expires class.

> Im not sure this method is warranted, nor does the EpochFormat enum. I only see one usage of it that essentially could be replaced with / 1000 (which was also the original code).
> 
>  I'd rather not introduce code we don't really use, even if it might seem sensible.